### PR TITLE
Remove extra newlines from formatting which resulted in necessary double newlines

### LINKED ISSUE #7186: Remove unnecessary newlines from output
The formatting code currently adds newlines unnecessarily so you get double and sometimes triple newlines that just take up screen space.  Affects all output types.

Steps to reproduce
------------------

```powershell
Get-Module
```

Expected behavior
-----------------

```none

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     0.0        build                               {Clear-PSRepo, Compress-TestContent, Convert-TxtResourceToXml, ConvertFrom-PesterLog...}
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty...}
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}
Script     2.0.0      PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler...}

PS /Users/steve/repos/PowerShell>
```

Actual behavior
---------------

```none

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     0.0        build                               {Clear-PSRepo, Compress-TestContent, Convert-TxtResourceToXml, ConvertFrom-PesterLog...}
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty...}
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}
Script     2.0.0      PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler...}


PS /Users/steve/repos/PowerShell>
```

Environment data
----------------

```powershell
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.3
PSEdition                      Core
GitCommitId                    v6.1.0-preview.3
OS                             Darwin 17.6.0 Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

When formatting, the `Group` start and end added unnecessary newlines which resulted in double newlines in many cases.  Formatting changes are not considered breaking.

```powershell
(get-module | out-string).replace("`n","\n`n")
```

Before:

```output
\n
ModuleType Version    Name                                ExportedCommands\n
---------- -------    ----                                ----------------\n
Manifest   6.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty...}\n
Manifest   6.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}\n
Script     2.0.0      PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler...}\n
\n
\n
```

After:

```output
\n             
ModuleType Version    Name                                ExportedCommands\n
---------- -------    ----                                ----------------\n
Manifest   6.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty...}\n
Manifest   6.1.0.0    Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object...}\n
Script     2.0.0      PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler...}\n
\n
```

There are still a few cases where the final output has two newlines (but not three anymore!).  This is a bit more complicated to fix and would need to track state that either a newline was the last thing output or the item is the last in a collection and doesn't need a separator.

Fix https://github.com/PowerShell/PowerShell/issues/7186

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by SteveL-MSFT** [comment]
Before:

```output


    Directory: /Users/steve/repos/PowerShell


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----           11/4/18  7:47 AM                assets


    Directory: /Users/steve/repos/PowerShell


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----          10/26/18 10:24 AM                demos


    Directory: /Users/steve/repos/PowerShell


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----          10/26/18 10:24 AM                docker


```

After:
```output


    Directory: /Users/steve/repos/PowerShell

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----           11/4/18  7:47 AM                assets

    Directory: /Users/steve/repos/PowerShell

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----          10/26/18 10:24 AM                demos

    Directory: /Users/steve/repos/PowerShell

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----          10/26/18 10:24 AM                docker

```

**Comment by SteveL-MSFT** [comment]
Here's example of `get-process pwsh | format-list`

Before:
```output


Id      : 1626
Handles : 0
CPU     : 5688.4889976
SI      : 422
Name    : pwsh

Id      : 18927
Handles : 0
CPU     : 470.8434028
SI      : 18920
Name    : pwsh

Id      : 18928
Handles : 0
CPU     : 745.9310174
SI      : 422
Name    : pwsh



```

After:
```output

Id      : 1626
Handles : 0
CPU     : 5688.0544032
SI      : 422
Name    : pwsh

Id      : 18927
Handles : 0
CPU     : 460.2073458
SI      : 18920
Name    : pwsh

Id      : 18928
Handles : 0
CPU     : 745.5975964
SI      : 422
Name    : pwsh


```

**Comment by SteveL-MSFT** [comment]
@iSazonov you are asking if it's ok to separate the groups with just a single newline vs two?  Seems fine to me and just as readable visually.  In the future, we should leverage vt100 as a way to provide visual indicators rather than rely on whitespace only.

**Review by iSazonov — APPROVED** [review]

File: test/powershell/Language/Classes/Scripting.Classes.Attributes.Tests.ps1 [hunk 1]
```diff
@@ -401,7 +401,7 @@ Describe 'ValidateSet support a dynamically generated set' -Tag "CI" {
                 Import-Module -Name $moduleFile -Force
                 Test-ValidateSet 'Hello' | Should -BeExactly 'Hello'
             } finally {
-                Remove-Module -Name $moduleFile -Force
+                Remove-Module -Name $moduleFile -Force -ErrorAction SilentlyContinue
```
Comment (anmenaga) [inline test/powershell/Language/Classes/Scripting.Classes.Attributes.Tests.ps1]: This is not related to this formatting code change, right?

**Review by anmenaga — APPROVED** [review]

File: test/powershell/Language/Classes/Scripting.Classes.Attributes.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply test/powershell/Language/Classes/Scripting.Classes.Attributes.Tests.ps1]: Correct.  Was running the tests locally and this line was outputting unncessarily

**Comment by adityapatwardhan** [comment]
@iSazonov  Reminder to add `CL-XXX` labels.
